README.md

```diff
@@ -1 +1,588 @@
+
+- [Tasks, Microtasks, Queues and Schedules](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/) by [Jake Archibald](https://twitter.com/jaffathecake)
+- [In The Loop](https://www.youtube.com/watch?v=cCOL7MC4Pl0) by [Jake Archibald](https://twitter.com/jaffathecake)
+- [JavaScript Visualized](https://dev.to/lydiahallie/javascript-visualized-promises-async-await-5gke#tasks) by [Lydia Hallie](https://twitter.com/lydiahallie)
+
+Fortunately, where we're headed, we don't need to understand all the ins and outs of the event loop, we're really only interested in _microtasks_ for our case. Consider the following piece of code (... that often does the rounds on Twitter polls, and has confused many a developer before). What do you think is the order the `console.log` statements are called?
+
+```js
+console.log(1);
+
+Promise.resolve().then(() => {
+  console.log(2);
+});
+
+console.log(3);
+```
+
+Here's the answer, the order of the resulting console logs of this code is:
+
+```js
+// 1
+// 3
+// 2
+```
+
+We'll have [Jake Archibald](https://twitter.com/jaffathecake) explain to us why exactly this happens:
+
+> The microtask queue is processed after callbacks as long as no other JavaScript is mid-execution, and at the end of each task. Any additional microtasks queued during microtasks are added to the end of the queue and also processed.
+
+If we go back to our code snippet, we can see that the following happens:
+
+- `console.log(1);` is called, and logged to the console
+- We schedule a new _microtask_ with `Promise.resolve().then(() => {})`, but at this point in time, JavaScript has not finished executing; it's not time to actually process microtasks just yet
+- JavaScript continues executing code, because we're not done yet
+- `console.log(3);` is called, and logged to the console
+- JavaScript has finished executing! The browser can now process any microtasks we may have
+- `console.log(2);` is called, and logged to the console.
+
+So why is this important? How does this allow us to use microtasks to _batch_ work? Let's take a look at a practical example:
+
+```js
+/* We declare and initialize a variable to keep track of whether or not an update has already been requested */
+let updateRequested = false;
+
+function scheduleUpdate() {
+  /** 
+   * In here, we need a check to see if an update is already previously requested.
+   * If an update already is requested, we don't want to do any unnecessary work!
+   */
+  if(!updateRequested) {
+
+    /* If no update has yet been requested, we set the `updateRequested` flag to `true` */
+    updateRequested = true;
+
+    /** 
+     * Since we now know that microtasks run after JavaScript has finished
+     * executing, we can use this knowledge to our advantage, and only set 
+     * the `updateRequested` flag back to `false` again once all the tasks 
```

Comment:
this comment is the heart of the matter. consider putting it in the post body below, not the snippet